From: Chad Monroe <chad@monroe.io>
Date: Mon, 24 Nov 2025 14:36:47 -0800
Subject: [PATCH] wifi-scripts: detect 320MHz EHT capability on 6GHz

Set max channel width to 320MHz if supported by hardware.

Signed-off-by: Chad Monroe <chad@monroe.io>
---
 src/usr/share/hostap/wifi-detect.uc |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/src/usr/share/hostap/wifi-detect.uc
+++ b/src/usr/share/hostap/wifi-detect.uc
@@ -164,7 +164,9 @@ function wiphy_detect() {
 				eht_phy_cap |= ift.eht_cap_phy[0];
 			}
 
-			if (band_name != "2G" &&
+			if (band_name == "6G" && (eht_phy_cap & 0x2))
+				band_info.max_width = 320;
+			else if (band_name != "2G" &&
 			    (he_phy_cap & 0x18) || ((band.vht_capa >> 2) & 0x3))
 				band_info.max_width = 160;
 			else if (band_name != "2G" &&
